canonical-web-and-design/maas-ui
# fix(ui): display cert and polling errors when adding an LXD KVM

## Done

- Update the add LXD host wizard to display errors when generating a cert or authenticating with LXD.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /LXD.
- Click Add LXD host.
- Enter a really long name (e.g. "thishosthasalongnamethatisallonewordonereallyreallyreallylongword").
- Enter a valid address and click "Next".
- An error should appear about the name being too long.
- Now enter a short name.
- Enter a valid address but change the port (e.g. "10.1.238.1:8442").
- Click "Next".
- Click "Check authentication".
- An error should appear about not connecting to LXD.
- Close the form and open it again.
- Enter a valid name and use the single-host address ("10.1.238.1:8443").
- Click "Next".
- Click "Check authentication".
- The new cert you generated won't have been added to the host so you should see the spinner, but no error should appear about the cert not being trusted (that error should be there in the pod.errors state).

## Fixes

Fixes: canonical-web-and-design/app-squad#483.
Fixes: canonical-web-and-design/maas-ui#3197.
